doc/javascripts/main.js

```diff
@@ -44,32 +44,17 @@ function initSidebar() {
     var href = link.href;
     if (href.lastIndexOf(filename) >= 0) {
       var parent = link.parentNode;
-      var p = document.createElement("p");
 
       while (link.childNodes.length > 0) {
         var child = link.childNodes[0];
         link.removeChild(child);
-        p.appendChild(child);
+        parent.appendChild(child);
```

Comment:
I can't say I'm excited about this uncommented JS code that does templating on the sidebar. @dwrensha any way I can convince you to add comments, as part of this pull request?